agents/agents-tools/src/commonMain/kotlin/ai/koog/agents/core/tools/Tool.kt

```diff
@@ -32,7 +31,7 @@ public interface DirectToolCallsEnabler
  * Represents a tool that, when executed, makes changes to the environment.
  */
 @Suppress("UNCHECKED_CAST", "unused")
-public abstract class Tool<TArgs : Tool.Args, TResult : ToolResult> {
+public abstract class Tool<TArgs : ToolArgs, TResult : ToolResult> {
```

Comment:
Please take a look at how `Tool` class was refactored. I've tried to remove methods used only for testing and God method `executeAndSerialize` as it was not useful anymore